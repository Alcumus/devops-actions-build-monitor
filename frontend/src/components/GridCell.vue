<template>
  <div
    :id="content.id"
    :key="content.id"
    class="grid-cell failure"
  >
    <div class="grid-cell-name">
      <a
        :id="content.id + '_url'"
        :href="content.url"
        target="_blank"
      >
        {{ content.name }}
      </a>
    </div>
  </div>
</template>

<script>
export default {
  name: "GridCell",
  props: {
    content: {
      type: Object,
      required: true
    }
  },
}
</script>

<style scoped>
.grid-cell-name {
  color: white;
  font-weight: bold;
  font-size: 14px;
  height: 100%;
  overflow: auto;
  display: inline-grid;
  align-items: center;
}

a {
  text-decoration: none;
  color: inherit;
}

.grid-cell {
  box-shadow: 5px 5px 10px #777;
  border-radius: 6px;
  height: 90px;
  border: 2px solid #000;
  padding: 5px;
}

.failure {
  background-color: #e23d2c;
}

</style>
