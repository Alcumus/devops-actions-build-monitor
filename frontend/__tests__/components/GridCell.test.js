import { describe, it, expect } from 'vitest';
import GridCell from '@/components/GridCell';
import { mount } from '../test-utils';

describe('<GridCell />', () => {
  it('should render grid cell', () => {
    const wrapper = mount(GridCell, {
      props: {
        content: {
          id: '1234',
          name: 'test name',
          url: 'https://test.com'
        }
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
