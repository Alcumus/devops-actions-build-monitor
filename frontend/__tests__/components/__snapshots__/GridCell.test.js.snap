// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GridCell /> > should render grid cell 1`] = `
<div class="v-card v-theme--light v-card--density-default v-card--variant-elevated grid-cell failure" id="1234" style="height: 90px;">
  <!---->
  <div class="v-card__loader">
    <div aria-hidden="true" aria-valuemax="100" aria-valuemin="0" class="v-progress-linear v-theme--light v-locale--is-ltr" role="progressbar" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);">
      <!---->
      <div class="v-progress-linear__background" style="width: 100%;"></div>
      <transition-stub appear="false" css="true" name="fade-transition" persisted="false">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class="v-card-text grid-cell-name pa-0 ma-0">
    <a aria-describedby="v-tooltip-0" href="https://test.com" id="1234_url" target="_blank">test name</a>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <!---->
  <!---->
  <span class="v-card__underlay"></span>
</div>
`;
